public/js/public.js

```diff
@@ -161,5 +161,37 @@
       Cookie.erase(my_gofc_object.cookie_name);
       jQuery('.GIGFILLIATE_PLACING_ORDER_FOR_CUSTOMER_DELETE').toast('show');
     }
+
+    $(".gfc_add_customer").on("click",function(e){
+      e.preventDefault();
+      e = $(this);
+      let email = null;
+      if(e.data("input_element")){
+        email = $(e.data("input_element")).val();
+        if(email.trim() == ""){
+          alert("Email is required.");
```

Comment:
You should just make the email field have the required attribute and validate email on input:
```
function validateEmail(email) {
    const re = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
    return re.test(String(email).toLowerCase());
}
```